python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -1897,3 +1906,70 @@ def unique_id_from_asset_and_check_keys(entity_keys: Iterable["EntityKey"]) -> s
     """
     sorted_key_strs = sorted(str(key) for key in entity_keys)
     return non_secure_md5_hash_str(json.dumps(sorted_key_strs).encode("utf-8"))[:8]
+
+
+def _thread_deps_to_node(
+    assets_def: AssetsDefinition, additional_deps: Mapping[AssetKey, AbstractSet[AssetDep]]
+) -> "AssetsDefinition":
+    from dagster._builtins import Nothing
+    from dagster._core.definitions.input import In
+
+    if not assets_def.is_executable:
+        return assets_def
+    if not isinstance(assets_def.node_def, OpDefinition):
+        raise DagsterInvalidDefinitionError("Cannot add dependencies to a graph-backed asset")
+    additional_input_names_by_key = {
```

Comment:
super confusing variable name since it is actually indexed by name

Reply:
oof. Yea I'll fix that